### PR TITLE
Haystack README

### PR DESCRIPTION
**Pull request by kylesayrs**
This readme originally was a part of https://github.com/neuralmagic/deepsparse/pull/597.

The readme specifies information about the embedding extraction pipeline, the two nodes, and the haystack pipeline.

### CODE REVIEW

File: integrations/haystack/README.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+```
+
+### Document Retrieval with DeepSparse ###
+To integrate with the DeepSparse Engine, simply replace your Haystack retriever node with an instance of a DeepSparse node.
+``` python3
+from deepsparse.transformers.haystack import DeepSparseEmbeddingRetriever
+
+retriever = DeepSparseEmbeddingRetriever(
+    document_store,
+    model_path="zoo:nlp/masked_language_modeling/bert-base/pytorch/huggingface/bookcorpus_wikitext/3layer_pruned90-none",
```
Comment (mgoin) [inline integrations/haystack/README.md:87]: is there a quantized mlm model we can use here? would definitely be faster and smaller

Comment (bfineran) [reply integrations/haystack/README.md:87]: we have a pruned, quantized distilbert
@kylesayrs please confirm this works
```suggestion
    model_path="zoo:nlp/masked_language_modeling/distilbert-none/pytorch/huggingface/wikipedia_bookcorpus/pruned80_quant-none-vnni",
```

Comment (kylesayrs) [reply integrations/haystack/README.md:87]: Confirmed to work

**Review by markurtz — DISMISSED** [review]

File: integrations/haystack/README.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+## Installation and Setup ##
+In order to ensure the proper installation, please use only install with python version 3.8.
+
+Install `farm-haystack`'s dependencies via deepsparse extras
+```bash
+pip install deepsparse[haystack]
+```
+
+After this is done, importing assets from `deepsparse.transformers.haystack` will trigger an auto-installation of Neural Magic's fork of `transformers` as well as `farm-haystack[all]==1.4.0`. These auto-installations can be controlled by setting the environment variables `NM_NO_AUTOINSTALL_TRANSFORMERS` and `NM_NO_AUTOINSTALL_HAYSTACK` respectively.
+
```
Comment (corey-nm) [inline integrations/haystack/README.md:15]: Does this need to be changed based on recent updates to how haystack is installed? We need them to do --no-dependencies right?

Comment (kylesayrs) [reply integrations/haystack/README.md:15]: In the source, haystack is still installed with no-dependencies. Are you suggesting that we note this in the docs?

Comment (corey-nm) [reply integrations/haystack/README.md:15]: Nevermind, just looked back at the PR I was thinking about. Looks good!

**Review by corey-nm — DISMISSED** [review]

File: integrations/haystack/README.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+|pruned80-vinni-untied|0.7867939539|98.82%|0.8460178556|99.15%|0.8629010872|99.50%|
+
+|MSMARCO Passage|MRR@10|Recall|Accuracy @10|Recall|Recall@20|Recall|Recall@100|Recall|Recall@200|Recall|
+|-|-|-|-|-|-|-|-|-|-|-|
+|base-none|0.3220429117|100.00%|0.6021489971|100.00%|0.6979942693|100.00%|0.8528653295|100.00%|0.8951289398|100.00%|
+|base-none-untied|0.3209568722|99.66%|0.5984240688|99.38%|0.6892550143|98.75%|0.8484240688|99.48%|0.8928366762|99.74%|
+|pruned90-none|0.3276589007|101.74%|0.6146131805|102.07%|0.7004297994|100.35%|0.8537249284|100.10%|0.8932664756|99.79%|
+|pruned90-none-untied|0.3093550166|96.06%|0.588252149|97.69%|0.6793696275|97.33%|0.8368194842|98.12%|0.8812320917|98.45%|
+|pruned80-vinni|0.3251235958|100.96%|0.6068767908|100.79%|0.6962750716|99.75%|0.8449856734|99.08%|0.8856733524|98.94%|
+|pruned80-vinni-untied|0.3124041479|97.01%|0.5918338109|98.29%|0.6802292264|97.45%|0.8319484241|97.55%|0.8780802292|98.10%|
```
Comment (mgoin) [inline integrations/haystack/README.md:296]:
```suggestion
pruned80-vnni|0.7847645429|98.30%|0.856232687|99.17%|0.8717451524|98.59%|
pruned80-vnni-untied|0.7817174515|97.92%|0.8509695291|98.56%|0.8717|98.59%|

|Model (TriviaNQ) |Accuracy @20|Recall|Accuracy @100|Recall|Accuracy @ 200|Recall|
|-|-|-|-|-|-|-|
|base-none|0.7961637055|100.00%|0.853266154|100.00%|0.8672323875|100.00%|
|base-none-untied|0.7943074339|99.77%|0.8503491558|99.66%|0.8661716609|99.88%|
|pruned90-none|0.7839653496|98.47%|0.8440731901|98.92%|0.8609564218|99.28%|
|pruned90-none-untied|0.782904623|98.33%|0.8435428268|98.86%|0.8594537258|99.10%|
|pruned80-vnni|0.7930699196|99.61%|0.8480509149|99.39%|0.8649341466|99.73%|
|pruned80-vnni-untied|0.7867939539|98.82%|0.8460178556|99.15%|0.8629010872|99.50%|

|MSMARCO Passage|MRR@10|Recall|Accuracy @10|Recall|Recall@20|Recall|Recall@100|Recall|Recall@200|Recall|
|-|-|-|-|-|-|-|-|-|-|-|
|base-none|0.3220429117|100.00%|0.6021489971|100.00%|0.6979942693|100.00%|0.8528653295|100.00%|0.8951289398|100.00%|
|base-none-untied|0.3209568722|99.66%|0.5984240688|99.38%|0.6892550143|98.75%|0.8484240688|99.48%|0.8928366762|99.74%|
|pruned90-none|0.3276589007|101.74%|0.6146131805|102.07%|0.7004297994|100.35%|0.8537249284|100.10%|0.8932664756|99.79%|
|pruned90-none-untied|0.3093550166|96.06%|0.588252149|97.69%|0.6793696275|97.33%|0.8368194842|98.12%|0.8812320917|98.45%|
|pruned80-vnni|0.3251235958|100.96%|0.6068767908|100.79%|0.6962750716|99.75%|0.8449856734|99.08%|0.8856733524|98.94%|
|pruned80-vnni-untied|0.3124041479|97.01%|0.5918338109|98.29%|0.6802292264|97.45%|0.8319484241|97.55%|0.8780802292|98.10%|
```

File: integrations/haystack/README.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+## Performance Evaluation ##
+Retrievers were also evaluated on their run time. This table compares the run time of generating query embeddings using `DenseEmbeddingRetriever` with Pytorch and `DeepSparseEmbeddingRetriever` with the DeepSparse Engine. Both retrievers were evaluated with the same 12 layer BERT model on the same hardware using CPU.
+
+|Number of Queries|DenseEmbeddingRetriever (sec)|DeepSparseEmbeddingRetriever (sec)|
+|-|-|-|
+|1|0.027|0.012|
+|100|1.37|0.96|
+|1,000|12.6|9.5|
```
Comment (mgoin) [inline integrations/haystack/README.md:305]: nit: would be best to run this with a quantized model

Comment (kylesayrs) [reply integrations/haystack/README.md:305]: Will add this in a followup PR

**Review by corey-nm — APPROVED** [review]

File: integrations/haystack/README.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
+# Haystack: Information Retrieval #
+The relevant features added as a part of the Haystack information retrieval integration are a [Haystack pipeline](/src/deepsparse/transformers/haystack/pipeline.py), an [embedding extraction pipeline](/src/deepsparse/transformers/pipelines/embedding_extraction.py), and two classes, [DeepSparseEmbeddingRetriever](/src/deepsparse/transformers/haystack/nodes.py) and [DeepSparseDensePassageRetriever](/src/deepsparse/transformers/haystack/nodes.py).
+
+These features allow a user to perform information retrieval tasks using the Haystack library as well as substitute in sparse retrieval nodes into their existing Haystack systems.
+
+## Installation and Setup ##
+In order to ensure the proper installation, please use only install with python version 3.8.
```
Comment (rahul-tuli) [inline integrations/haystack/README.md:7]:
```suggestion
In order to ensure the proper installation, please install with python version 3.8.
```

**Review by bfineran — APPROVED** [review]

File: integrations/haystack/README.md [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+from haystack.document_stores import InMemoryDocumentStore
+
+document_store = InMemoryDocumentStore(
+    similarity="cosine",
+    embedding_dim=768,
+    use_gpu=False
+)
+```
+
+Next, create a retriever. The retriever is houses the deep model and is responsible for, given a document or query, generating an embedding such that query embeddings have a high similarity to their relevant document embeddings.
```
Comment (rahul-tuli) [inline integrations/haystack/README.md:33]:
```suggestion
Next, create a retriever. The retriever houses the deep model and is responsible for, given a document or query, generating an embedding such that query embeddings have a high similarity to their relevant document embeddings.
```

File: integrations/haystack/README.md [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+    pooling_strategy="cls_token",
+)
+document_store.update_embeddings(retriever)
+pipeline = DocumentSearchPipeline(retriever)
+
+results = pipeline.run(query="How is everybody feeling?", params={"Retriever": {"top_k": 1}})
+```
+
+## Haystack Pipeline ##
+The haystack pipeline is a non traditional pipeline which constructs Haystack nodes which are used for document retrieval or any other Haystack task. Said another way, this pipeline provides an API for constructing a document_store, retriever, and pipeline like the workflow described in [Document Retrieval with Haystack](#Document-Retrieval-with-Haystack)
```
Comment (rahul-tuli) [inline integrations/haystack/README.md:171]:
```suggestion
The haystack pipeline is a non traditional pipeline which constructs Haystack nodes used for document retrieval or any other Haystack task. Said another way, this pipeline provides an API for constructing a document_store, retriever, and pipeline like the workflow described in [Document Retrieval with Haystack](#Document-Retrieval-with-Haystack)
```

**Review by rahul-tuli — APPROVED** [review]

**Comment by kylesayrs** [comment]
Will make a followup PR with more typo changes and performance testing

**Pull request merged**